datacube_stats/main.py

```diff
@@ -821,6 +825,35 @@ def boundary_polygon_from_file(filename):
     return boundary_polygon
 
 
+def filter_time_from_sensor(source_spec, time_period):
+    """
+    Override date ranges if sensor specific time is within the time_period range
+    """
+    ep_range = time_period
+    if source_spec.get('time'):
```

Comment:
For the future prefer to extract value of interest from a collection and then use it, rather than reading again and again at different call sites. In this case

```python
time = source_spec.get('time')
if time is not None:
    # do stuff with time variable
```

This reduces probability of mis-typing key 'time' during later accesses and makes it easier to change code when key name is renamed or value is preprocessed in some  way.

In fact this is exactly what you want to do: preprocess time variable by parsing it with strptime.

```python
time = source_spec.get('time')
if time is not None:
    time = [datetime.strptime(v, "%Y-%m-%d") for v in time]
```

Repeating calls to `strptime` with repeated format string is very noisy, hard to read and likely to make a mistake in one of those lines without it being spotted, hard to change format etc.